src/main/java/part2/Service/ShopConnection.java

```diff
@@ -0,0 +1,149 @@
+package part2.Service;
+
+import lombok.extern.slf4j.Slf4j;
+import part2.Product.Product;
+import part2.Product.User;
+
+import java.sql.*;
+import java.util.HashMap;
+import java.util.List;
+
+@Slf4j
+public class ShopConnection {
+    private final String url = "jdbc:mysql://localhost:3306/test?useSSL=false";
+    private final String login = "root";
+    private final String password = "root";
+    private static ShopConnection shopConnection;
+    private Connection connection;
+
+    private ShopConnection() {
+        try {
+            this.connection = DriverManager.getConnection(url, login, password);
+        } catch (SQLException ex) {
+            log.error("Database Connection Creation Failed : " + ex.getMessage());
```

Comment:
getMessage/getCause not always represents some data could be null. so use just e)